Fe forum page #32

jonquag opened the pull request:
Here is an updated look at the forum page. 

Having a few styling issues but I think that may sort itself out once we get some merges completed. Looking for some advice on the styling for the inline part at the bottom of the post content.

Commit:
Fixed merge

Commit:
Combined login, register, logout routers into authRouter

Commit:
Changed route names on frontend to match auth routes

jonquag committed:
Added default Person icon

Added a default icon to the user img if none exists.
Styling changes

Commit:
Integrated logout with frontend

jonquag committed:
More styling changes

The use form is now styled properly.
Groups no longer bring s asecond Nav Bar.
Nav Bar nowhs link styling when active.

amha19 committed:
Refactored fe/be to fetch all the user info

Added active styling to navbar.
Edited get profile route.

jonquag committed:
added submit functionality

Submit changes now sends all data to the DB.

jonquag committed:
update pre-merge

amha19 committed:
Refactored profile and user routes

amha19 committed:
Fixed minor userinfo integration

Commit:
Handle sign in without university

Commit:
removed login and register route files

Commit:
Create user before save

amha19 committed:
Fixed fetching and storing user info

amha19 committed:
Removed console logs

jonquag committed:
added placeholders for imputs

 Added plaveholders for inputs and removed useEffect from imports

jonquag committed:
Added forum cards and passing them mock data

This is my first draft of the forum page using cards. I'm having some trouble with in-line styling for the text on the bottom of the card. I think the alignment will be fixable once we get some of the branches merged.

jonquag commented:
![Forum page](https://user-images.githubusercontent.com/67391359/105865543-0a3c8f80-5fc1-11eb-9a12-ffeb61d92cea.png)

jonquag committed:
Updated styling for spacing and fixed inline issues

Fixed styling issues. Added Divider. Button has temporary onClick functioality.

TODO: still has responsiveness issues.

jonquag commented:
![Forum page update 2](https://user-images.githubusercontent.com/67391359/105875848-4d503000-5fcc-11eb-8622-9af894727c01.png)

jonquag committed:
Added button theme

Added a theme for all buttons to be styled with the blue gradient, no text tansform and the white lettering.

jonquag committed:
updated theme buttons

Went through the code and updated theme buttons.  took out some replicated styling nested into components.

jonquag commented:
I have updated the button theme throughout the file.
to apply the styling to any buttons made add these attributes to your button element:

variant="text"
color="primary"

you may also still add styling to this button further (or to overwrite) for width and necessary padding through your own:

className={classes.button}

jessdesr commented:
Do you know why so many files were modified in this PR?

jonquag commented:
> Do you know why so many files were modified in this PR?

does this have to do with this branch being originally the demo branch?

jessdesr commented:
> > Do you know why so many files were modified in this PR?
> 
> does this have to do with this branch being originally the demo branch?

Yeah, it would.  You'll most likely want to rebase off of dev.  This a bit more of an experienced git practice, so I'll outline it for you:

1) Checkout dev and pull to get latest changes
2) Checkout this branch
3) Command: `git rebase dev`
4) If there are any conflicts while you're rebasing, just message me and we can work through it together

jonquag committed:
Merge branch 'dev' into FE-Forum-Page-#32

jonquag committed:
Merge branch 'dev' into FE-Forum-Page-#32

jonquag committed:
Updated theme buttons

jonquag committed:
Added two more theme buttons

Added two more theme buttons. Fixed styling on the login and sign up page buttons. Added the theme for the "send" button for the chats page.

jonquag committed:
Deleted some test buttons

timegambit commented on client/src/components/Forum/PostCard.js:
```diff
@@ -0,0 +1,80 @@
+    const classes = useStyles();
+
+    return (
+        <Card className={classes.root}>
```
I think this component needs a little more room for long text. In the image below the "Qualification" on the Google analytics post is right at the border. Maybe just adding a paddingRight would work. 

![Screen Shot 2021-01-28 at 9 10 46 AM](https://user-images.githubusercontent.com/45152761/106150663-8bbf2980-6149-11eb-9562-b24f7074210e.png)

Also I was wondering how you feel about making this a Paper component with an elevation. I think it looks more like the mockup. And I think there's a way to customize the shadow of the paper to give it that light blueish look. What is looks like using Paper with elevation=5:

![Screen Shot 2021-01-28 at 9 13 17 AM](https://user-images.githubusercontent.com/45152761/106151082-fd977300-6149-11eb-84cb-5c31cbb0fd1e.png)

jonquag replied:
Good spot on the right padding. I agree with you. I also thought the text size in the sketch file of 22 might be a bit large.  

The Paper idea is a great idea with the elevation. I'll look into this today.

amha19 replied:
The branch I am checking has the right padding and the cards look great. 
If you try using 'rem' for font sizes let me know if you are getting any errors.

timegambit commented on client/src/components/Forum/ForumContent.js:
```diff
@@ -0,0 +1,85 @@
+                    <Button
+                        className={classes.button}
+                        onClick={addPost}
+                        variant="text"
```
I think variant="text" by default, so you don't need to include this line on buttons.

jonquag replied:
> I think variant="text" by default, so you don't need to include this line on buttons.

variant="text"
color="primary"

Are setting the button to the override in theme.js
Not sure if leaving it out would render that override button, but I can take a look

timegambit replied:
I tested it out and it looked like it did!

jonquag committed:
Added paper to PostCard.js and removed unecessary button stylings.

jonquag commented:
![forum post cards with paper](https://user-images.githubusercontent.com/67391359/106167633-6509ee80-615b-11eb-90e7-cf3eb93361ae.png)

Here's an updated look at the forum PostCard.js with paper and the paddingRight added. Great spot,  @timegambit

timegambit commented:
> ![forum post cards with paper](https://user-images.githubusercontent.com/67391359/106167633-6509ee80-615b-11eb-90e7-cf3eb93361ae.png)
> 
> Here's an updated look at the forum PostCard.js with paper and the paddingRight added. Great spot, @timegambit

Looks great Jon!

jessdesr commented:
I think I might set the width of the posts to be proportionate to the size of the container, with a larger maxWidth set and maybe a degree of padding?  For example, check the padding in the mockup (even though the mockups are a pretty weird aspect ratio) and see if you can set width to 100% and padding to match the mockup.  

You can test how it would show up on larger screens by using the magnification function in chrome.   We don't want it to get ridiculously long though so set a maxWidth with your best judgement

jessdesr reviewed (approved):
Looks good, nice work on the buttons

jonquag reviewed (commented):
> I think I might set the width of the posts to be proportionate to the size of the container, with a larger maxWidth set and maybe a degree of padding? For example, check the padding in the mockup (even though the mockups are a pretty weird aspect ratio) and see if you can set width to 100% and padding to match the mockup.
> 
> You can test how it would show up on larger screens by using the magnification function in chrome. We don't want it to get ridiculously long though so set a maxWidth with your best judgement

I can look into this.

ethanrosenberg reviewed (approved):
Great work Jon! The paper addition looks really good

jonquag committed:
Updated post width

Updated post max width and added padding.
